source/slang/slang-emit-metal.cpp

```diff
@@ -1182,6 +1182,11 @@ void MetalSourceEmitter::emitSimpleTypeImpl(IRType* type)
                 m_writer->emit(" object_data");
                 m_writer->emit("*");
                 break;
+            default:
+                // For unrecognized address spaces, default to constant for uniform parameters
```

Comment:
If this is `UserPointer` address space, we should make sure they are translated to `Device` address space during legalization.